alphadia/outputtransform.py

```diff
@@ -869,13 +873,33 @@ def _build_run_stat_df(
         if "mobility_fwhm" in channel_df.columns:
             base_dict["fwhm_mobility"] = np.mean(channel_df["mobility_fwhm"])
 
-        with open(optimization_manager_path, "rb") as f:
-            optimization_manager = pickle.load(f)
+        if os.path.exists(optimization_manager_path):
+            optimization_manager = manager.OptimizationManager(
+                path=optimization_manager_path
+            )
+
+            base_dict["ms2_error"] = optimization_manager.ms2_error
+            base_dict["ms1_error"] = optimization_manager.ms1_error
+            base_dict["rt_error"] = optimization_manager.rt_error
+            base_dict["mobility_error"] = optimization_manager.mobility_error
+
+        else:
+            logger.warning(f"Error reading optimization manager for {raw_name}")
+            base_dict["ms2_error"] = np.nan
+            base_dict["ms1_error"] = np.nan
+            base_dict["rt_error"] = np.nan
+            base_dict["mobility_error"] = np.nan
+
+        if os.path.exists(timing_manager_path):
+            timing_manager = manager.TimingManager(path=timing_manager_path)
 
-        base_dict["ms2_error"] = optimization_manager.ms2_error
-        base_dict["ms1_error"] = optimization_manager.ms1_error
-        base_dict["rt_error"] = optimization_manager.rt_error
-        base_dict["mobility_error"] = optimization_manager.mobility_error
+            base_dict["optimization_duration"] = timing_manager.optimization["duration"]
```

Comment:
if `TimingManager` had this info stored in an actual dictionary (e.g. `timings`), you could do smth like

```python
for key in timing_manager.timings:
  base_dict[f"duration_{key}"] = timing_manager.timings[key]
```
thus avoiding an explicit coupling of this to the individual timings
(cf also the comment below)